perspective.el

```diff
@@ -482,7 +508,8 @@ If NORECORD is non-nil, do not update the
   (if (null name) (setq name (persp-prompt (and (persp-last) (persp-name (persp-last))))))
   (if (and (persp-curr) (equal name (persp-name (persp-curr)))) name
     (let ((persp (gethash name (perspectives-hash))))
-      (set-frame-parameter nil 'persp--last (persp-curr))
+      (when (and (persp-curr) (not (persp-killed (persp-curr))))
```

Comment:
`(not (persp-killed (persp-curr)))` makes the `persp-let-frame-parameters` in `persp-kill` unnecessary.